utils/rpcd-mod-poe/Makefile

```diff
@@ -0,0 +1,41 @@
+# This is free software, licensed under the GNU General Public License v2.
+# See /LICENSE for more information.
+#
+
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=rpcd-mod-poe
+PKG_VERSION:=0.1
+PKG_RELEASE:=1
+PKG_LICENSE:=GPL-2.0
+
+include $(INCLUDE_DIR)/package.mk
+
+define Package/rpcd-mod-poe
+  SECTION:=utils
+  CATEGORY:=Base system
+  TITLE:=OpenWrt ubus PoE interface
+  MAINTAINER:=Paul Spooren <mail@aparcar.org>
+  DEPENDS:=+rpcd +kmod-i2c-gpio-custom
+endef
+
+define Package/rpcd-mod-poe/description
+	Expose ubus procedures to query, enable and disable the per-port PoE power
+	on Ubiquity devices and similar platforms.
```

Comment:
the name seems both overly specific when it's just doign gpio toggling, and overly generic, when it's only handling PoE for thigns using this particular gpio_switch config for ubiquity.  DO you really just want a generic gpio control?

Reply:
I want to power cycle PoE ports, nothing more. Please help me find a clean solution for that. I'm happy to use device tree or whatever else.

Reply:
For real PoE there are vendor-specific hacks, see
https://docs.cumulusnetworks.com/display/DOCS/Power+over+Ethernet+-+PoE
for inspiration.

Maybe the information about the mapping of PoE-GPIOs to Ethernet ports can somehow be described in device-tree in a way the kernel wouldn't care but make it possible for userspace to retrieve that information? That would already make it possible to describe eg. the PoE-passthrough of simple ubnt devices.
I reckon for a clean solution for ER-X we will have to wait for DSA finding use for MT7621 -- the MT7630 driver was recently extended to [support the MT7621](https://github.com/torvalds/linux/commit/ddda1ac116c852bb969541ed53cffef7255c4961#diff-5aed940c7b18a8002cacb9dce5bd5c1f) upstream, so all we will have to do is to convert the existing swconfig LAN/WAN setup into DSA for all boards...

Reply:
How much work is it? More like ~2 month or 2 years?